jujugui/static/gui/src/app/app.js

```diff
@@ -1251,6 +1252,20 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Updates the db's environment name when the model name is changed
+      in the deployment panel.
+
+      @method _updateModelName
+    */
+    _updateModelName: function(e) {
```

Comment:
This appears to contain code specific to the deployment flow. Is there a particular reason you decided to do this here instead of passing a reference to `db.environment` into the DF and setting the the name there? I might have missed the reason...